llama_index/vector_stores/simple.py

```diff
@@ -36,6 +38,26 @@
 MMR_MODE = VectorStoreQueryMode.MMR
 
 
+def _build_metadata_filter_fn(
+    metadata_lookup_fn: Callable[[str], Mapping[str, Any]],
+    metadata_filters: Optional[MetadataFilters] = None,
+) -> Callable[[str], bool]:
+    """Build metadata filter function."""
+    filter_list = metadata_filters.filters if metadata_filters else []
+    if not filter_list:
+        return lambda _: True
+
+    def filter_fn(node_id: str) -> bool:
+        metadata = metadata_lookup_fn(node_id)
+        for filter_ in filter_list:
+            metadata_value = metadata.get(filter_.key, None)
+            if metadata_value is None or metadata_value != filter_.value:
+                return False
+        return True
+
+    return filter_fn
+
+
 @dataclass
 class SimpleVectorStoreData(DataClassJsonMixin):
```

Comment:
I don't think there's an issue here, but just as a sanity check, you you try loading an index that was saved before these changes with the current code? Just to test backwards compatibility

Reply:
tested -- it works fine